extensions/azdata/src/azdata.ts

```diff
@@ -156,21 +156,23 @@ export class AzdataTool implements azdataExt.IAzdataApi {
 						memoryLimit?: string,
 						memoryRequest?: string,
 						noWait?: boolean,
-					}): Promise<azdataExt.AzdataOutput<void>> => {
+					},
+					additionalEnvVars?: { [key: string]: string }
+				): Promise<azdataExt.AzdataOutput<void>> => {
 					const argsArray = ['arc', 'sql', 'mi', 'edit', '-n', name];
 					if (args.coresLimit) { argsArray.push('--cores-limit', args.coresLimit); }
 					if (args.coresRequest) { argsArray.push('--cores-request', args.coresRequest); }
 					if (args.memoryLimit) { argsArray.push('--memory-limit', args.memoryLimit); }
 					if (args.memoryRequest) { argsArray.push('--memory-request', args.memoryRequest); }
 					if (args.noWait) { argsArray.push('--no-wait'); }
-					return this.executeCommand<void>(argsArray);
+					return this.executeCommand<void>(argsArray, additionalEnvVars);
 				}
 			}
 		}
 	};
 
-	public login(endpoint: string, username: string, password: string): Promise<azdataExt.AzdataOutput<void>> {
-		return this.executeCommand<void>(['login', '-e', endpoint, '-u', username], { 'AZDATA_PASSWORD': password });
+	public login(endpoint: string, username: string, password: string, additionalEnvVars: { [key: string]: string } = {}): Promise<azdataExt.AzdataOutput<void>> {
+		return this.executeCommand<void>(['login', '-e', endpoint, '-u', username], Object.assign(additionalEnvVars, { 'AZDATA_PASSWORD': password }));
```

Comment:
Do `Object.assign({}, additionalEnvVars, { ... })` instead - we don't want to modify the additionalEnvVars object being passed in

Reply:
done.